conda-incubator/conda-store
# DEV - Update conda-store build hook

## Description

Recent updates to conda-store-ui broke the already brittle build process of `conda-store-server` and `conda-store-ui`. This PR aims to make this more robust and remove some hacky bits we have relied on until now.

Note that the UI counterpart is at https://github.com/conda-incubator/conda-store-ui/pull/418
so both PRs _must_ be merged for the build and release to work.



## Pull request checklist




## Additional information

This PR not only improves our build process but makes it so that one can build `conda-store-server` with a local copy repo/copy of `conda-store-ui` (vs only a released version).
To test these two PRs together:

1. Check out https://github.com/conda-incubator/conda-store-ui/pull/418 in your local clone of `conda-store-ui`
2. Bundle the package as if for a release with `run webpack:prod bundle`
3. Check out this PR on your local clone of `conda-store` and cd to `conda-store-server`
4. Build `conda-store-server` with your local copy of `conda-store-ui` with `export LOCAL_UI=<full path to the root the conda-store-ui repo>  && hatch build --clean`
6. Run `conda-store-server` in standalone mode:

```
# quick way to verify the build
conda-store-server --standalone
```

You should see the dev version of the UI 

![image](https://github.com/user-attachments/assets/3ac8a59c-fdcc-405e-9117-aa282273a300)
